client/components/domains/register-domain-step/index.jsx

```diff
@@ -113,6 +121,7 @@ const RegisterDomainStep = React.createClass( {
 	},
 
 	componentWillMount: function() {
+		searchCount = 0; // reset the counter
```

Comment:
I think `lastSearchTimestamp` should be reset to null here too - otherwise, when the user navigates away from the view and comes back, the first search will be reported with a very large time diff.